controller/tests/test_volunteer_controller.py

```diff
@@ -0,0 +1,109 @@
+import unittest
+import controller.volunteer_controller as vc
+from models.camp import Camp
+from models.plan import Plan
+
+
+class TestVolunteerController(unittest.TestCase):
+
+    def test_edit_firstname(self):
+        volunteer = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                        phone='+447519953189', camp=Camp(name='UCL'))
+        volunteer = vc.edit_firstname(volunteer, 'Yun-Tzu')
+        self.assertEqual(volunteer.firstname, 'Yun-Tzu')
+
+    def test_edit_firstname_invalid_input(self):
+        with self.assertRaises(vc.ControllerError):
+            volunteer = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                            phone='+447519953189', camp=Camp(name='UCL'))
+            vc.edit_firstname(volunteer=volunteer, firstname='Y')
+
+    def test_edit_lastname(self):
+        volunteer = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                        phone='+447519953189', camp=Camp(name='UCL'))
+        volunteer = vc.edit_lastname(volunteer=volunteer, lastname='Yinnn')
+        self.assertEqual(volunteer.lastname, 'Yinnn')
+
+    def test_edit_lastname_invalid_input(self):
+        with self.assertRaises(vc.ControllerError):
+            volunteer = vc.create_volunteer(username='yunsy', password='root', firstname='Yunsy', lastname='Yin',
+                                            phone='+447519953189', camp=Camp(name='UCL'))
```

Comment:
It would be better to put create volunteer outside of self.assertRaises, so we are sure that the error comes from the exit not the create
Same for the other tests

Reply:
Thanks for pointing that out! I'll fix them.